lib/manager/npm/post-update/index.ts

```diff
@@ -336,9 +337,71 @@ async function resetNpmrcContent(
   }
 }
 
+// istanbul ignore next
+async function updateYarnOffline(
+  lockFileDir: string,
+  localDir: string,
+  updatedArtifacts: UpdatedArtifacts[]
+): Promise<void> {
+  try {
+    const resolvedPaths: string[] = [];
+    const yarnrc = await getFile(upath.join(lockFileDir, '.yarnrc'));
+    const yarnrcYml = await getFile(upath.join(lockFileDir, '.yarnrc.yml'));
+
+    if (yarnrc) {
```

Comment:
I'd recommend to switch those two blocks (first check for `.yarnrc.yml`, then for `.yarnrc`). In my work repo for instance, we have both a `.yarnrc` and a `.yarnrc.yml`, with the first one pointing to a binary explaining that the global Yarn isn't recent enough because it doesn't read `.yarnrc.yml`, and needs to be upgraded.

Reply:
Is there a minimum version of Yarn v1 that supports "bootstrapping" Yarn v2?

Reply:
Yarn 1 reading paths from `.yarnrc.yml` started with 1.18.0. Before that it was still possible, but you had to put the right path in both files.

Reply:
Updated the PR.